Better lint execution for deleted and modified files

sebmck opened the pull request:
Addresses some issues listed in #389.

 - Added `WorkerFileNotFound` error that survives being transferred over the worker bridge. We use this to swallow errors with `MasterRequest#requestAllowDeletion`. Initially I had a `memoryFs.exists` call just in the lint callback. That however will still allow for race conditions between when `exists` is called and the actual `readFile` call. This way there's no real opportunity for race conditions.
- Single lint runs will now include files modified (or added!) while executing.

cc @yassere

yassere commented:
Including modified/added files into an already-running lint is very cool!

The `WorkerFileNotFound` error will be great for preventing issues when a file is deleted while that file’s request is being processed.

I think that there often won’t be a `WorkerFileNotFound` error for `requestAllowDeletion` to catch because when `requestWorkerLint` is called with a nonexistent path, it attempts an error-throwing cache lookup before it sends anything over the bridge.

Should Cache#get also throw a `FileNotFound` error that `requestAllowDeletion` can catch? I think it would also need to be handled within DependencyGraph for LintRunner#runGraph to work.

sebmck commented:
Right you are!

I renamed `WorkerFileNotFound` to just `FileNotFound` and made it emit in a bunch of places that would have previously been hard errors. Now the `MemoryFileSystem#getMtime` call will throw a `FileNotFound` which is called by `Cache#get`. I also added a `FileNotFound.allowMissing` option to `DependencyGraph#seed`.

Now we have a pretty basic reusable primitive for gracefully handling deleted/incorrect filenames.
